### PR TITLE
RUMM-948 Reinforce first party host rules and add  exception

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

- reinforces the first party hosts validation rules to not accept top - level - domain - only hosts (e.g. `invalid-host`)
- add special wildcard rule for `localhost`

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: dd-sdk-android/src/main/kotlin/com/datadog/android/core/configuration/Configuration.kt [hunk 1]
```diff
@@ -496,6 +496,9 @@ internal constructor(
                     }
                 } else if (it.matches(validHostNameRegEx)) {
                     it
+                } else if (it == "localhost") {
```
Comment (mariusc83) [inline dd-sdk-android/src/main/kotlin/com/datadog/android/core/configuration/Configuration.kt:499]: should we go with `no case` comparison here ?

**Review by ncreated — APPROVED** [review]
👌 👌 👌

File: dd-sdk-android/src/main/kotlin/com/datadog/android/core/configuration/Configuration.kt (hunk 1 above)
Comment (xgouchet) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/core/configuration/Configuration.kt:499]: Yes that sounds like a good idea

File: dd-sdk-android/src/main/kotlin/com/datadog/android/core/configuration/Configuration.kt [hunk 2]
```diff
@@ -467,7 +467,7 @@ internal constructor(
             "^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\\.)" +
                 "{3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$|" +
                 "^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\\-]*[a-zA-Z0-9])\\.)" +
-                "*([A-Za-z]|[A-Za-z][A-Za-z0-9-]*[A-Za-z0-9])$"
+                "+([A-Za-z]|[A-Za-z][A-Za-z0-9-]*[A-Za-z0-9])$"
```
Comment (xgouchet) [inline dd-sdk-android/src/main/kotlin/com/datadog/android/core/configuration/Configuration.kt:470]: While you're there, maybe you could make this a bit more readable. Among other things, the cardinality modifiers should be at the end of a line, not the beginning. E.g.: `"(…){3}" + "…"` or  `"(…)\\.+" + "(…)$"`

**Review by xgouchet — APPROVED** [review]

**Pull request merged**